src/handlers/pull_requests_assignment_update.rs

```diff
@@ -0,0 +1,68 @@
+use crate::github::retrieve_pull_requests;
+use crate::jobs::Job;
+use anyhow::Context as _;
+use async_trait::async_trait;
+use tokio_postgres::Client as DbClient;
+
+pub struct PullRequestAssignmentUpdate;
+
+#[async_trait]
+impl Job for PullRequestAssignmentUpdate {
+    fn name(&self) -> &'static str {
+        "pull_request_assignment_update"
+    }
+
+    async fn run(&self, ctx: &super::Context, _metadata: &serde_json::Value) -> anyhow::Result<()> {
+        let db = ctx.db.get().await;
+        let gh = &ctx.github;
+
+        tracing::trace!("starting pull_request_assignment_update");
+
+        // delete everything before populating
+        init_table(&db).await?;
+
+        let rust_repo = gh.repository("rust-lang/rust").await?;
+        let prs = retrieve_pull_requests(&rust_repo, &gh).await?;
+
+        // populate the table
+        for (assignees, pr_num) in prs {
+            for assignee in assignees {
+                let assignee_id = assignee.id.expect("checked");
+                ensure_team_member(&db, assignee_id, &assignee.login).await?;
+                create_team_member_workqueue(&db, assignee_id, pr_num).await?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// Truncate the review prefs table
+async fn init_table(db: &DbClient) -> anyhow::Result<u64> {
+    let res = db.execute("TRUNCATE review_prefs;", &[]).await?;
+    Ok(res)
+}
+
+/// Add a new user (if not existing)
+async fn ensure_team_member(db: &DbClient, user_id: i64, username: &str) -> anyhow::Result<u64> {
+    let q = "
+INSERT INTO users (user_id, username) VALUES ($1, $2)
+ON CONFLICT DO NOTHING";
+    let rec = db
+        .execute(q, &[&user_id, &username])
+        .await
+        .context("Insert user DB error")?;
+    Ok(rec)
+}
+
+/// Create a team member work queue
+async fn create_team_member_workqueue(
+    db: &DbClient,
+    user_id: i64,
+    pr: i32,
```

Comment:
I feel like we should aggregate by user *first*, rather than doing a db request for each PR.